### PR TITLE
vlc handler: do timings / locks correctly; other cleanup

### PR DESCRIPTION
**Pull request by renekliment**
This should fix the silly bug we have.
Could be reproduced like this:
1. Ask Alexa to play something from TuneIn
2. Trigger her while she plays something
3. The whole thing gets frozen.

Turns out asynchronous things together with reusing the VLC player is not a totally trivial task. It's fixed now hopefully.

This also:
- refactors the locks a bit (hey, there's a `.clear()` method, great!)
- deletes files after the upload of the recording, not before so it doesn't waste time, because storage I/O can be slow - hello SD cards ... this should help where the temporary directory is not in the RAM
- moves the small pause between tracks where it belongs, so you shouldn't experience the 0.5s delay when playing just one file or something
- cleans up a few debug messages

It's a little messy, meaning it's multiple things in one PR, but I didn't want to open several PRs for many tiny things so I hope it's okay.

@alexa-pi/coreteam @dgonano Please test.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -524,31 +532,22 @@ def loop():
 				player.play_speech(resources_path + 'alexayes.mp3')
 				time.sleep(0.5)
```
Comment (dgonano) [inline src/main.py]: What is this delay for after the "yes?" I feel like it's an unnecessary pause before starting recording.

File: src/main.py [hunk 2]
```diff
@@ -524,31 +532,22 @@ def loop():
...
 
-		# do the following things if either the button has been pressed or the trigger word has been said
-		if debug:
-			print("detected the edge, setting up audio")
-
 		# To avoid overflows close the microphone connection
 		inp.close()
 
+		silence_listener(VAD_THROWAWAY_FRAMES)
+		alexa_speech_recognizer()
```
Comment (dgonano) [inline src/main.py]: Nice work putting this before the speech recognition. I had a note in my file to move it also.

File: src/main.py (hunk 1 above)
Comment (renekliment) [inline src/main.py]: Nothing anymore. I temporarily made the playback call non-blocking and had to wait here and then forgot about this. Great catch! I believe we're now back at the trigger-to-listen speed as before. It's been really slow as you've pointed out somewhere. And now we're back baby!

**Comment by dgonano** [comment]
Having a few issues with radio still. 
1. "tripple j" station plays well an can now be interrupted. I believe it should resume playing after the current task is done? it just stops and doesn't continue. 
2. "Nove 106.9" doesn't work. say "Play nova one thousand and sixty nine radio" and Alexa acknowledges it, says it playing, but no music plays.

**Comment by renekliment** [comment]
@dgonano 

- ad 1: Great to hear that! Not with the current code. We need to 
    1. discuss that (pause/resume vs background music - probably only with PA or custom handlers) 
    2. have a look at the _playBehaviour_ attribute from AVS that tells what to do with current queue
    3. think about the current code to allow for the above by not calling `.stop()` every time to make sure everything will be fine
- ad 2: Works for me. Can you post output? It takes time for the TuneIn code to decode the actual streaming URL. Also, TuneIn sometimes seems to work / not work by a region. Weird stuff.

**Comment by dgonano** [comment]
1. Okay, no worries. something to work on I suppose. 
2. 
```Processing Request Response...
JSON String Returned: {"messageBody": {"directives": [{"namespace": "Speaker", "name": "SetMute", "payload": {"mute": false}}, {"namespace": "SpeechSynthesizer", "name": "speak", "payload": {"contentIdentifier": "amzn1.as-ct.v1.Domain:Application:Music#ACRI#DeviceTTSRendererV4_662fcbc6-dbfe-4965-b6ed-235ea6044e91", "audioContent": "cid:DeviceTTSRendererV4_662fcbc6-dbfe-4965-b6ed-235ea6044e91_489585439"}}, {"namespace": "Speaker", "name": "SetMute", "payload": {"mute": false}}, {"namespace": "AudioPlayer", "name": "play", "payload": {"playBehavior": "REPLACE_PREVIOUS", "audioItem": {"streams": [{"progressReport": {"progressReportDelayInMilliseconds": 0, "progressReportIntervalInMilliseconds": 900000}, "streamUrl": "http://opml.radiotime.com/Tune.ashx?id=s87120&formats=aac,mp3&partnerId=4JqugguZ&serial=AEOK4VMXAH7UA7B3KRR7GPYG63IA&latlon=40.75,-73.98333", "offsetInMilliseconds": 0, "expiryTime": "2016-12-10T14:14:32+0000", "streamId": "amzn1.as-ct.v1.Dee-Domain-Music#ACRI#url#ACRI#50257d7c-a1a5-4689-96e6-6a1c276f32cc:1", "progressReportRequired": true}], "audioItemId": "amzn1.as-ct.v1.Dee-Domain-Music#ACRI#url#ACRI#50257d7c-a1a5-4689-96e6-6a1c276f32cc:1"}, "navigationToken": "amzn1.as-ct.v1.Dee-Domain-Music#ACRI#url#ACRI#50257d7c-a1a5-4689-96e6-6a1c276f32cc:1"}}]}, "messageHeader": {}}
Indicate Stop Recording
Calling URL: http://localhost:8080/alexapi?action=AVSSTATUS&status=idle
Response: {"status":"success"}
Play_Audio Request for: file:///tmp/AlexaPi-runtime-bjULz4/DeviceTTSRendererV4_662fcbc6-dbfe-4965-b6ed-235ea6044e91_489585439.mp3
Player State: State.Opening
Player State: State.Playing
Indicate Start Playing
Calling URL: http://localhost:8080/alexapi?action=AVSSTATUS&status=playback
Response: {"status":"success"}
Sending MM Heatbeat
Response: {"status":"success"}
Player State: State.Ended
Indicate Stop Playing
Calling URL: http://localhost:8080/alexapi?action=AVSSTATUS&status=idle
TUNE IN URL = http://opml.radiotime.com/Tune.ashx?id=s87120&formats=aac,mp3&partnerId=4JqugguZ&serial=AEOK4VMXAH7UA7B3KRR7GPYG63IA&latlon=40.75,-73.98333
```
I forgot to mention that it locks up also, you can't invoke it anymore so you have to crash the program and start it again. 
I've listened to both this station and triple j in the alexia-sample-app so they should work fine

**Comment by renekliment** [comment]
@dgonano
So since the TuneIn bug is in the external library / its handling and is unrelated to the original bug, I'm asking for approval to merge this.

We need to fix the TuneIn stuff separately as it's an issue on its own and it will require some amount of testing. Also, apart from fixing the stuff, we should also handle the library failure so it doesn't block our code. The constructor timeout obviously doesn't work on a global scale / at all.

**Comment by dgonano** [comment]
@renekliment Yeah that's fine

**Comment by renekliment** [comment]
Great. Thanks.

**Pull request merged**